modules/xfem/src/userobjects/MeshCut2DRankTwoTensorNucleation.C

```diff
@@ -84,13 +100,125 @@ MeshCut2DRankTwoTensorNucleation::doesElementCrack(
 
   if (tensor_quantity > average_threshold)
   {
-    const Point elem_center(_current_elem->vertex_average());
     does_it_crack = true;
+
+    const Point elem_center(_current_elem->vertex_average());
     Point crack_dir = point_dir.cross(Point(0, 0, 1));
-    RealVectorValue point_0 = elem_center - _nucleation_length / 2.0 * crack_dir.unit();
-    RealVectorValue point_1 = elem_center + _nucleation_length / 2.0 * crack_dir.unit();
+
+    // get max element length to use for ray length
+    std::unique_ptr<const Elem> edge;
+    Real circumference = 0;
+    for (const auto e : _current_elem->edge_index_range())
+    {
+      _current_elem->build_edge_ptr(edge, e);
+      circumference += edge->length(0, 1);
+    }
+
+    // Temproraries for doing things below
+    Real intersection_distance;
+
+    Point point_0;
+    Point point_1;
+    bool is_point_0_on_external_boundary = false;
+    bool is_point_1_on_external_boundary = false;
+    for (const auto e : _current_elem->edge_index_range())
+    {
+      _current_elem->build_edge_ptr(edge, e);
+      const auto & edge0 = edge->point(0);
+      const auto & edge1 = edge->point(1);
+      if (lineLineIntersect2D(
+              elem_center, -crack_dir, circumference, edge0, edge1, point_0, intersection_distance))
+        is_point_0_on_external_boundary = (_current_elem->neighbor_ptr(e) == nullptr);
+      else if (lineLineIntersect2D(elem_center,
+                                   crack_dir,
+                                   circumference,
+                                   edge0,
+                                   edge1,
+                                   point_1,
+                                   intersection_distance))
+        is_point_1_on_external_boundary = (_current_elem->neighbor_ptr(e) == nullptr);
+    }
+
+    // bisect_length is the length of a cut that goes across a single elment
```

Comment:
These comments are helpful. I find 'bisect_length' a little confusing, because it sounds like we might be bisecting the cut length or something. I guess it's referring to bisecting the element. I think it might be more clear if it's called something like 'traverse_length'.